src/transformers/processing_utils.py

```diff
@@ -1102,10 +1123,42 @@ def apply_chat_template(
                     "or provide a chat template as an argument. See "
                     "https://huggingface.co/docs/transformers/main/en/chat_templating for more information."
                 )
-        return self.tokenizer.apply_chat_template(
-            conversation, chat_template=chat_template, tokenize=tokenize, **kwargs
+
+        prompt = self.tokenizer.apply_chat_template(
+            conversation, chat_template=chat_template, tokenize=False, return_dict=False, **kwargs
         )
 
+        def parse_conversation(conversation: Union[List[Dict[str, str]]]):
+            images, videos = [], []
+            for message in conversation:
+                visuals = [content for content in message["content"] if content["type"] in ["image", "video"]]
+                for vision_info in visuals:
+                    if vision_info["type"] == "image" and "image" in vision_info:
+                        images.append(load_image(vision_info["image"]))
+                    elif vision_info["type"] == "video" and "video" in vision_info:
+                        videos.append(load_video(vision_info["video"]))
```

Comment:
I mean, here we call `load_video` without passing `backend`, so I'm curious how can anyone specify it for the chat template preprocessing. Or is it expected to use `backend` only for "manual" `load_video` call?

Reply:
ohh, i see, for chat template I think we can leave the default value as it is the one which works in all cases. I left the backend for manual `load_video`, yes, when one wants to build their own collator etc.

Do you think we need to enable `backend` when applying chat template? I guess it doesn't hurt, the only reason I didn't add was because I didn't want to add more and more kwargs to the method hehe

Reply:
I see your point, I suppose it would be nice at least to specify `backend` explicitly here. Then we can check if it will be requested :)

Also, there is no way to specify `num_frames` right now

Reply:
oh yeah, `num_frames` is actually super important. I'll make changes

Reply:
```suggestion
                        videos.append(load_video(vision_info["video"], num_frames=num_frames, backend=backend))
```